src/operator/nn/mkldnn/mkldnn_deconvolution.cc

```diff
@@ -127,11 +148,21 @@ static mkldnn::convolution_backward_weights::primitive_desc GetDeconvBwdWeights(
   if (param.stride.ndim() == 2) {
     strides[0] = param.stride[0];
     strides[1] = param.stride[1];
+  } else if (param.stride.ndim() == 1) {
+    strides[0] = param.stride[0];
+    strides[1] = param.stride[0];
+  } else {
+    LOG(FATAL) << "Unsupported stride dim";
   }
   mkldnn::memory::dims padding{0, 0};
   if (param.pad.ndim() == 2) {
     padding[0] = param.pad[0];
     padding[1] = param.pad[1];
+  } else if (param.pad.ndim() == 1) {
+    padding[0] = param.pad[0];
+    padding[1] = param.pad[0];
```

Comment:
Are you sure this is right way to handle 1D padding, striding, etc?

Reply:
Yes. If just input one integer as stride, it will be expand to a list/array for strides in different directions. You can find keras as a simple example: https://github.com/keras-team/keras/blob/master/keras/layers/convolutional.py#L108
But I am not sure mxnet supports stride and padding inputs with other dims besides 2D. If just 2D is supported, we can throw error when just one integer is inputted.

Reply:
Yes. If just one integer is inputted as stride, it will be expanded into a list/array for striding in different directions. You can take keras as a simple example: https://github.com/keras-team/keras/blob/master/keras/layers/convolutional.py#L108
But I am not sure mxnet can support stride and padding with other dims besides 2D. If can't, we can throw error if there is just one integer.